internal/batches/batch_spec.go

```diff
@@ -107,6 +108,21 @@ type Group struct {
 func ParseBatchSpec(data []byte, features featureFlags) (*BatchSpec, error) {
 	var spec BatchSpec
 	if err := yaml.UnmarshalValidate(schema.BatchSpecJSON, data, &spec); err != nil {
+		if multiErr, ok := err.(*multierror.Error); ok {
+			var newMultiError *multierror.Error
+
+			for _, e := range multiErr.Errors {
+				// In case of `name` we try to make the error message more user-friendly.
+				if strings.Contains(e.Error(), "name: Does not match pattern") {
```

Comment:
If the error message changes sometime in the future, this will stop working. Would it make sense to add a test to catch that kind of regression?

Reply:
Good point! I was unsure about to test this (or whether I should) but a tiny test that makes sure this behaviour stays sounds good to me.